src/cert_context.rs

```diff
@@ -228,7 +270,48 @@ impl CertContext {
         }
     }
 
-    fn get_encoded_bytes(&self) -> &[u8] {
+    /// Returns the valid uses for this certificate
+    pub fn valid_uses(&self) -> io::Result<ValidUses> {
+        unsafe {
+            let mut buf_len = 0;
+            let ok = crypt32::CertGetEnhancedKeyUsage(self.0, 0, ptr::null_mut(), &mut buf_len);
```

Comment:
Seems like you should be able to just stack allocate a `CERT_ENHKEY_USAGE` in and avoid the double call and vector.

Reply:
The docs don't really call it out, but I verified that the struct has a variable sized trailer.